src/main/java/uk/gov/hmcts/reform/unspec/stateflow/StateFlowEngine.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.unspec.stateflow;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.unspec.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.unspec.model.CaseData;
+import uk.gov.hmcts.reform.unspec.stateflow.model.State;
+
+import static uk.gov.hmcts.reform.unspec.stateflow.StateFlowEngine.FlowState.CLAIM_ISSUED;
+import static uk.gov.hmcts.reform.unspec.stateflow.StateFlowEngine.FlowState.CLAIM_STAYED;
+import static uk.gov.hmcts.reform.unspec.stateflow.StateFlowEngine.FlowState.DRAFT;
+import static uk.gov.hmcts.reform.unspec.stateflow.StateFlowEngine.FlowState.EXTENSION_REQUESTED;
+import static uk.gov.hmcts.reform.unspec.stateflow.StateFlowEngine.FlowState.EXTENSION_RESPONDED;
+import static uk.gov.hmcts.reform.unspec.stateflow.StateFlowEngine.FlowState.FULL_DEFENCE;
+import static uk.gov.hmcts.reform.unspec.stateflow.StateFlowEngine.FlowState.RESPONDED_TO_CLAIM;
+import static uk.gov.hmcts.reform.unspec.stateflow.StateFlowEngine.FlowState.SERVICE_ACKNOWLEDGED;
+import static uk.gov.hmcts.reform.unspec.stateflow.StateFlowEngine.FlowState.SERVICE_CONFIRMED;
+import static uk.gov.hmcts.reform.unspec.stateflow.utils.PredicateUtils.claimantConfirmService;
+import static uk.gov.hmcts.reform.unspec.stateflow.utils.PredicateUtils.claimantIssueClaim;
+import static uk.gov.hmcts.reform.unspec.stateflow.utils.PredicateUtils.claimantRespondToDefence;
+import static uk.gov.hmcts.reform.unspec.stateflow.utils.PredicateUtils.claimantRespondToRequestForExtension;
+import static uk.gov.hmcts.reform.unspec.stateflow.utils.PredicateUtils.defendantAcknowledgeService;
+import static uk.gov.hmcts.reform.unspec.stateflow.utils.PredicateUtils.defendantAskForAnExtension;
+import static uk.gov.hmcts.reform.unspec.stateflow.utils.PredicateUtils.defendantRespondToClaim;
+import static uk.gov.hmcts.reform.unspec.stateflow.utils.PredicateUtils.schedulerStayClaim;
+
+@Component
+@RequiredArgsConstructor
+public class StateFlowEngine {
```

Comment:
Could we have this in its own package? The `uk.gov.hmcts.reform.unspec.stateflow` should be for the `framework`. I would create a new package and have in there:
- StateFlowEngine
- FlowState
- FlowPredicate

Reply:
Agreed. I am extracting them in next PR